test/test_python_dispatch.py

```diff
@@ -97,6 +97,25 @@ def capture_logs() -> Iterator[List[str]]:
     finally:
         logger.removeHandler(handler)
 
+class TestPythonRegistration(TestCase):
+    def test_override_cpu_sum(self) -> None:
+        m = torch._C._dispatch_library("FRAGMENT", "aten", "")
+
+        run = [False]
+
+        def my_sum(*args, **kwargs):
+            run[0] = True
+            return args[0]
+
+        m.impl("sum", "CPU", my_sum)
```

Comment:
Hmm adding a python API to register ops is going to surface existing bugs in the implementaiton of some functionality keys like Conjugation, Negation, ZeroTensor ... since many of the ops currently skip the respective fallback for these keys when they are set. Even if we add a per backend registration after Brian's PR, this issue won't be resolved. Maybe we can automatically register the fallback for these functionality keys in these cases?

Reply:
Don't we already? A backend fallback is supposed to apply universally.

Reply:
not if some ops are registered as fallthrough

Reply:
Yeah it's true. Automatically reregistering the fallback key could work but it feels a bit too magic for my taste. Need to think about this...

Reply:
we could add a warning when we automatically reregister the fallback key, and let the user explicitly register it as fallthrough if they wish to do so.

Reply:
Well, for the custom op use case in functorch, this is not a big deal since there won't be a fallthrough registration. So maybe let's focus on that use case first.